crates/mcp-server/src/lib.rs

```diff
@@ -19,9 +19,10 @@ pub use router::Router;
 #[pin_project]
 pub struct ByteTransport<R, W> {
     #[pin]
-    reader: R,
+    reader: BufReader<R>,
     #[pin]
     writer: W,
+    buf: Vec<u8>,
```

Comment:
nit: if my understanding of the issue is correct, we don't need to keep this as a member and clear it below? seems simpler to instantiate a new one in each poll next instead. but let me know if i am missing something